onnxruntime/core/providers/dml/dml_provider_factory.cc

```diff
@@ -55,6 +63,11 @@ std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_DML(ID
   return std::make_shared<onnxruntime::DMLProviderFactory>(dml_device, cmd_queue);
 }
 
+void DmlConfigureProviderFactoryDefaultRoundingMode(IExecutionProviderFactory* factory, AllocatorRoundingMode rounding_mode) {
+  auto dml_prvider_factory = static_cast<DMLProviderFactory*>(factory);
+  dml_prvider_factory->SetDefaultRoundingMode(rounding_mode);
+}
+
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_DML(int device_id) {
   ComPtr<IDXGIFactory4> dxgi_factory;
```

Comment:
>ComPt [](start = 2, length = 5)

This code to create a dml device should be reused by the winml adapter...

Reply:
for a future refactor

---
In reply to: [370304546](https://github.com/microsoft/onnxruntime/pull/2891#discussion_r370304546) [](ancestors = 370304546)